Packs/Absolute/Integrations/Absolute/Absolute.py

```diff
@@ -893,6 +1063,86 @@ def get_device_location_command(args, client) -> CommandResults:
 
 
+''' EVENT COLLECTOR '''
+
+
+def fetch_events(client: ClientV3, fetch_limit: int, last_run: Dict[str, Any]) -> tuple[List[Dict[str, Any]], Dict[str, Any]]:
+    next_page_token = last_run.get('next_page_token', '')
+    last_run_latest_events_id = last_run.get('latest_events_id', [])
+    last_end_date = last_run.get('end_date', '')
+    last_start_date = last_run.get('start_date', '')
+    end_date = datetime.utcnow()
+
+    # start_date = datetime.strptime(last_start_date, "%Y-%m-%d %H:%M:%S") if next_page_token else (
+    #     datetime.strptime(last_end_date, "%Y-%m-%d %H:%M:%S") if last_end_date else end_date - timedelta(minutes=1))
+
+    # TODO: remove this temp start_date
+    start_date = datetime.strptime(last_start_date, "%Y-%m-%d %H:%M:%S") if next_page_token else (
+        datetime.strptime(last_end_date, "%Y-%m-%d %H:%M:%S") if last_end_date else end_date - timedelta(minutes=7))
+    # TODO: remove this temp start_date
+
+    demisto.debug(f'Starting new fetch: {fetch_limit=}, {start_date=}, {end_date=}, {next_page_token=}, {last_run=}')
+    all_events, next_page_token = client.fetch_events_with_pagination(fetch_limit, next_page_token, start_date, end_date)
+    if not all_events:
+        return [], {
+            'next_page_token': '',
+            'start_date': start_date.strftime("%Y-%m-%d %H:%M:%S"),
+            'end_date': end_date.strftime("%Y-%m-%d %H:%M:%S"),
+            'latest_events_id': []}
+
+    events, latest_events_id = handle_duplication_and_add_time_field_to_events_and_get_latest_events(all_events,
+                                                                                                     last_run_latest_events_id)
+    demisto.debug(f'fetch_events: {latest_events_id=}')
+    return events, {
+        'next_page_token': next_page_token,
+        'start_date': start_date.strftime("%Y-%m-%d %H:%M:%S"),
+        'end_date': end_date.strftime("%Y-%m-%d %H:%M:%S"),
+        'latest_events_id': latest_events_id}
+
+
+def handle_duplication_and_add_time_field_to_events_and_get_latest_events(events: List[Dict[str, Any]],
+                                                                          last_run_latest_events_id: List[str],
+                                                                          should_get_latest_events: bool = True) -> tuple[
+    List[Dict[str, Any]], List[str]]:
+
+    demisto.debug("Handle duplicate events, adding _time field to events and optionally getting the latest events id")
+    earliest_event_time = events[0].get('eventDateTimeUtc')
+    latest_event_time = events[-1].get('eventDateTimeUtc')
+    latest_events_id = []
+    filtered_events = []
+    for event in events:
+        event_time = event.get('eventDateTimeUtc')
+        # handle duplication
+        if event_time == earliest_event_time and event.get('id') in last_run_latest_events_id:
+            continue
```

Comment:
```suggestion
        if event.get('id') in last_run_latest_events_id:
            continue
```
This condition should be enough to handle duplication

Reply:
i did it in purpose to reduce the times of checking if id exist in the list. 
think about it, if there is 1000 events, and last_run_latest_events_id hold some id's (maybe could be also large number), why to pass with all the 1000 over the list?